fairlearn/metrics/_metric_frame.py

```diff
@@ -171,6 +179,13 @@ class MetricFrame:
         a nested dictionary, with the first set of string keys identifying the
         metric function name, with the values being the string-to-array-like dictionaries.
 
+    streaming : bool
+        If True, will modify the behavior to accept the accumulation of values before computing
+        the score. One can append data with ``add_data``. The user can't supply data in the
+        constructor if ``streaming=True``.
+
+        **Note** currently, it only stores the values before computing the metric.
```

Comment:
I'm not quite sure that this note is supposed to mean.

Reply:
In #655, we mentioned that in the future we would like to support object that have an "update function. Like [Keras Stateful Metrics](https://keras.io/api/metrics/) or [BaaL Metrics](https://github.com/ElementAI/baal/blob/master/src/baal/utils/metrics.py#L22).

In this PR, we simply accumulate the predictions and we run the metrics on all values.

Reply:
Isn't that always going to be required?

Reply:
Yeah I guess, unless we have multiple metrics object at the same time for each combination of sensitive attribute.

I'll remove for now and we can decide what to do later.